actionpack/lib/action_dispatch/journey/parser.rb

```diff
@@ -56,43 +56,43 @@ def parse_or(lhs)
       end
 
       def parse_expression
-        if @next_token.first == :STAR
+        if @next_token == :STAR
           parse_star
-        elsif @next_token.first == :LPAREN
+        elsif @next_token == :LPAREN
           parse_group
         else
           parse_terminal
         end
       end
 
       def parse_star
-        node = Star.new(Symbol.new(@next_token.last, Symbol::GREEDY_EXP))
+        node = Star.new(Symbol.new(@scanner.last_string, Symbol::GREEDY_EXP))
         advance_token
         node
       end
 
       def parse_group
         advance_token
         node = parse_expressions
-        if @next_token.first == :RPAREN
+        if @next_token == :RPAREN
           node = Group.new(node)
           advance_token
           node
         else
-          raise "error"
+          raise ArgumentError, "missing right parenthesis."
         end
       end
 
       def parse_terminal
-        node = case @next_token.first
+        node = case @next_token
         when :SYMBOL
-          Symbol.new(@next_token.last)
+          Symbol.new(@scanner.last_string)
         when :LITERAL
-          Literal.new(@next_token.last)
+          Literal.new(@scanner.last_literal)
         when :SLASH
-          Slash.new(@next_token.last)
+          Slash.new("/")
```

Comment:
Related to this: this file doesn't have the "# frozen_string_literal: true" pragma, is that on purpose? Or it doesn't; matter much?

Reply:
This is due to this file is excluded for rubocop target.

https://github.com/rails/rails/blob/ac104520ee8518441bb4f8c2c7835d0bb8fd6800/.rubocop.yml#L14-L18

I'll fix rubocop offences by #52654.